src/components/frame/Extensions/pages/ExtensionDetailsPage/partials/ExtensionDetailsHeader/hooks.tsx

```diff
@@ -85,53 +86,109 @@ export const useInstall = (extensionData: AnyExtensionData) => {
     );
   }, []);
 
-  const handleInstallSuccess = useCallback(async () => {
-    setIsLoading(true);
-    setWaitingForActionConfirmation(true);
+  const showInitialiseErrorToast = useCallback(() => {
+    toast.error(
+      <Toast
+        type="error"
+        title={{ id: 'extensionEnable.toast.title.error' }}
+        description={{ id: 'extensionEnable.toast.description.error' }}
+      />,
+    );
+  }, []);
 
-    try {
-      await waitForDbAfterExtensionAction({
-        method: ExtensionMethods.INSTALL,
-        refetchExtensionData,
-      });
+  const showSetUserRolesErrorToast = useCallback(() => {
+    toast.error(
+      <Toast
+        type="error"
+        title={{ id: 'extensionSetUserRoles.toast.title.error' }}
+        description={{ id: 'extensionSetUserRoles.toast.description.error' }}
+      />,
+    );
+  }, []);
 
-      toast.success(
-        <Toast
-          type="success"
-          title={{ id: 'extensionInstall.toast.title.success' }}
-          description={{
-            id: 'extensionInstall.toast.description.success',
-          }}
-        />,
-      );
+  const handleInstallSuccess = useCallback(
+    async ({
+      initialiseTransactionFailed,
+      setUserRolesTransactionFailed,
+    }: {
+      initialiseTransactionFailed?: boolean;
+      setUserRolesTransactionFailed?: boolean;
+    }) => {
+      setIsLoading(true);
+      setWaitingForActionConfirmation(true);
+
+      try {
+        await waitForDbAfterExtensionAction({
+          method: ExtensionMethods.INSTALL,
+          refetchExtensionData,
+          initialiseTransactionFailed,
+          setUserRolesTransactionFailed,
+        });
+
+        toast.success(
+          <Toast
+            type="success"
+            title={{ id: 'extensionInstall.toast.title.success' }}
+            description={{
+              id: 'extensionInstall.toast.description.success',
+            }}
+          />,
+        );
```

Comment:
@iamsamgibbs if there could be some errors while either initialising the transaction or setting the user roles, does it make sense to refetch the extension action and show the success toast?

Reply:
In the case where the initialise transaction fails, I think you might be right that we don't need to refetch the extension data.

In the case where the setUserRole transaction fails, we still want to refetch the extension data to check that the extension has been enabled still.

As for the success toast, it's a bit of a confusing ones as technically the install was successful, but the initalise / setUserRoles transactions failed, so I've opted to still show the install success but also show a failure toast for initalise / setUserRoles.